server/src/main/java/org/candlepin/service/ProductServiceAdapter.java

```diff
@@ -39,6 +39,13 @@
     Product getProductById(Owner owner, String id);
 
     /**
+     * Query a specific product by its string ID.
+     * @param id product id
+     * @return specific product
+     */
+    Product getProductById(String id);
+
```

Comment:
Not sure about this one-off method on the adapter. The original goal was to remove the adapter. I wonder if it would make sense to perhaps create a new method that is targeted at the dev sku. That way IT could isolate the lookup if required and trim down the adapter to that method later on, instead of completely removing it.

The fact that this feature doesn't work 'Out Of The Box' without a backing subscription source is bothersome. The 'shared product' concept here makes it difficult I know, since candlepin doesn't have that any more. Perhaps out of the box the feature targets an owner, but production targets the 'shared' product (ignores the owner arg during lookup). I'll have to think about this some more. Please do the same.